dask/dataframe/shuffle.py

```diff
@@ -785,25 +785,18 @@ def collect(p, part, meta, barrier_token):
 
 
 def set_partitions_pre(s, divisions, ascending=True, na_position="last"):
-    try:
-        if ascending:
-            partitions = divisions.searchsorted(s, side="right") - 1
-        else:
-            partitions = len(divisions) - divisions.searchsorted(s, side="right") - 1
-    except TypeError:
-        # `searchsorted` fails if `s` contains nulls and strings
-        partitions = np.empty(len(s), dtype="int32")
-        not_null = s.notna()
-        if ascending:
-            partitions[not_null] = divisions.searchsorted(s[not_null], side="right") - 1
-        else:
-            partitions[not_null] = (
-                len(divisions) - divisions.searchsorted(s[not_null], side="right") - 1
-            )
+    partitions = np.zeros_like(s)
```

Comment:
Whether or not we opt for the changes made in this PR, we should probably make sure to initialize `partitions` using an `np.*_like` function if we can't create it from `searchsorted`, otherwise we will end up with a NumPy array that causes failures when we try to index it with a cuDF series.